mountpoint-s3/tests/fuse_tests/mod.rs

```diff
@@ -145,6 +146,44 @@ mod mock_session {
         (mount_dir, session, test_client)
     }
 
+    /// Create a FUSE mount backed by a mock object client, with caching, that does not talk to S3
+    pub fn new_with_cache<Cache>(
+        cache: Cache,
+    ) -> impl FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox)
+    where
+        Cache: DataCache + Send + Sync + 'static,
+    {
+        |test_name, test_config| {
+            let mount_dir = tempfile::tempdir().unwrap();
+
+            let bucket = "test_bucket";
+            let prefix = if test_name.is_empty() {
+                test_name.to_string()
+            } else {
+                format!("{test_name}/")
+            };
+
+            let client_config = MockClientConfig {
+                bucket: bucket.to_string(),
+                part_size: test_config.part_size,
+            };
+            let client = Arc::new(MockClient::new(client_config));
+            let runtime = ThreadPool::builder().pool_size(1).create().unwrap();
+            let prefetcher = caching_prefetch(cache, runtime, test_config.prefetcher_config);
+            let session = create_fuse_session(
+                client.clone(),
+                prefetcher,
+                bucket,
+                &prefix,
+                mount_dir.path(),
+                test_config.filesystem_config,
+            );
+            let test_client = create_test_client(client, &prefix);
+
+            (mount_dir, session, test_client)
+        }
+    }
+
```

Comment:
It's more complicated because they need 2 different types of prefetchers. We may need to pass a "prefetcher creator" function. Not sure it's worth to avoid the code duplication at this time.

Reply:
Can't be done due to prefetcher type, no worries.